tests/test_4_immut.py

```diff
@@ -90,3 +91,40 @@ def test_mitochondria():
                        np.array([0, 1, 1, 2], dtype=np.float32))
 
     assert_equal(len(mito_root[1].children), 0)
+
+
+def test_vasculature():
+    morpho = vasculature.VasculatureMorphology(os.path.join(_path, "h5/vasculature.h5"))
```

Comment:
Test also with an empty morphology, a cyclic one, any weird case you can think of.